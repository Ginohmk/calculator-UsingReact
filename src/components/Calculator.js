import React, { useState } from 'react';
import calculate from '../logic/calculate';
import Button from './Button';
import Display from './display';

export default function Calculator() {
  const [state, setState] = useState({
    total: '0',
    operation: null,
    next: null,
  });

  const test = (val) => {
    const result = calculate(state, val);
    if (result.next === null && result.total === null) {
      result.total = '0';
    }
    setState(result);
  };

  const { total, operation, next } = state;

  return (
    <div className="App" data-testid="calculate-data">
      <div>
        <h2>Lets do some math!</h2>
      </div>
      <div className="container">
        <Display
          prevOperand={total}
          operation={operation || ''}
          curOperand={next || ''}
        />
        <div className="input">
          <div className="section-button">
            <Button
              perform={() => test('AC')}
              title="AC"
              className="operation"
            />
            <Button
              perform={() => test('+/-')}
              title="+/-"
              className="operation"
            />
            <Button perform={() => test('%')} title="%" className="operation" />
            <Button
              perform={() => test('รท')}
              title="รท"
              className="operation colored"
            />
          </div>

          <div className="section-button">
            <Button perform={() => test('7')} title="7" className="operation" />
            <Button perform={() => test('8')} title="8" className="operation" />
            <Button perform={() => test('9')} title="9" className="operation" />
            <Button
              perform={() => test('x')}
              title="x"
              className="operation colored"
            />
          </div>
          <div className="section-button">
            <Button perform={() => test('4')} title="4" className="operation" />
            <Button perform={() => test('5')} title="5" className="operation" />
            <Button perform={() => test('6')} title="6" className="operation" />
            <Button
              perform={() => test('-')}
              title="-"
              className="operation colored"
            />
          </div>

          <div className="section-button">
            <Button perform={() => test('1')} title="1" className="operation" />
            <Button perform={() => test('2')} title="2" className="operation" />
            <Button perform={() => test('3')} title="3" className="operation" />
            <Button
              perform={() => test('+')}
              title="+"
              className="operation colored"
            />
          </div>

          <div className="section-button">
            <Button
              perform={() => test('0')}
              title="0"
              className="operation zero"
            />
            <Button perform={() => test('.')} title="." className="operation" />
            <Button
              perform={() => test('=')}
              title="="
              className="operation colored"
            />
          </div>
        </div>
      </div>
    </div>
  );
}
