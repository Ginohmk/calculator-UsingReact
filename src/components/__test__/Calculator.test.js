import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../Calculator';

describe('TEst for Home component', () => {
  test('Home snapShot test', () => {
    const tree = renderer.create(<Calculator />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
