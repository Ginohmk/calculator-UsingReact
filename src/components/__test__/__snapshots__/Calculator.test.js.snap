// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Calculate component Home snapShot test 1`] = `
<div
  className="App"
  data-testid="calculate-data"
>
  <div>
    <h2>
      Lets do some math!
    </h2>
  </div>
  <div
    className="container"
  >
    <div
      className="output"
    >
      <p
        data-testid="display-data"
      >
         0
      </p>
    </div>
    <div
      className="input"
    >
      <div
        className="section-button"
      >
        <button
          className="operation"
          data-testid="AC"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="operation"
          data-testid="+/-"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="operation"
          data-testid="%"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="operation colored"
          data-testid="รท"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </div>
      <div
        className="section-button"
      >
        <button
          className="operation"
          data-testid="7"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="operation"
          data-testid="8"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="operation"
          data-testid="9"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="operation colored"
          data-testid="x"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="section-button"
      >
        <button
          className="operation"
          data-testid="4"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="operation"
          data-testid="5"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="operation"
          data-testid="6"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="operation colored"
          data-testid="-"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="section-button"
      >
        <button
          className="operation"
          data-testid="1"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="operation"
          data-testid="2"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="operation"
          data-testid="3"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="operation colored"
          data-testid="+"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="section-button"
      >
        <button
          className="operation zero"
          data-testid="0"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="operation"
          data-testid="."
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="operation colored"
          data-testid="="
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
